backend/plugin/setup.go

```diff
@@ -0,0 +1,55 @@
+package plugin
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"os"
+
+	hclog "github.com/hashicorp/go-hclog"
+)
+
+// SetupPluginEnvironment will read the environment variables and apply the
+// standard environment behavior.  As the SDK evolves, this will likely change!
+func SetupPluginEnvironment(pluginID string) hclog.Logger {
```

Comment:
I don't want to expose this to plugin developers. If we want to add it now it should automatically apply when you do `backend.Serve`. This is related to some other changes I would like to do regarding serving of plugins.

Reply:
Do abstract classes exist?  can you create the logger on startup also?

I am struggling with having 4 variations (and soon 8!) of creating loggers and setting up pprof -- I don't care at all about the details, I just want whatever we decide is the best approach to be applied to everything

Reply:
I think what Marcus suggest is to always call it from https://github.com/grafana/grafana-plugin-sdk-go/blob/master/backend/serve.go#L17

Reply:
can we add that now?  or wait?

I still like having a function that reads the environment and gives me a logger -- for now it can do the pprof setup, when you want to add that to Serve then we can add it later (or if you want to do it now)

The nice part is that we won't have to update the plugins if we add this utility now

Reply:
I think your right. Let's add this and perfect it later.